app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,114 @@
+    {
+        $this->wikiDomain = $wikiDomain;
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $mwPod = $kubernetesClient->pods()->setFieldSelector([
+            'status.phase' => 'Running'
+        ])->setLabelSelector([
+            'app.kubernetes.io/name' => 'mediawiki',
+            'app.kubernetes.io/component' => 'app-backend'
+        ])->first();
+
+        if (!$mwPod) {
+            $this->fail(
+                new \RuntimeException(
+                    'Unable to find a running MediaWiki pod in the cluster, '.
+                    'cannot continue.'
+                )
+            );
+            return;
+        }
+
+        $mwPod = $mwPod->toArray();
+
+        $jobSpec = new KubernetesJob([
+            'metadata' => [
+                'generateName' => 'run-all-mw-jobs-'
+            ],
+            'spec' => [
+                'ttlSecondsAfterFinished' => 60 * 60,
+                'template' => [
+                    'metadata' => [
+                        'name' => 'run-all-mw-jobs'
+                    ],
+                    'spec' => [
+                        'containers' => [
+                            0 => [
+                                'name' => 'run-all-mw-jobs',
+                                'image' => $mwPod['spec']['containers'][0]['image'],
+                                'env' => array_merge(
+                                    $mwPod['spec']['containers'][0]['env'],
+                                    [['name' => 'WBS_DOMAIN', 'value' => $this->wikiDomain]]
+                                ),
+                                'command' => [
+                                    0 => 'bash',
+                                    1 => '-c',
+                                    2 => <<<'CMD'
+                                    JOBS_TO_GO=1
+                                    while [ "$JOBS_TO_GO" != "0" ]
+                                    do
+                                        echo "Running 1000 jobs"
+                                        php w/maintenance/runJobs.php --maxjobs 1000
+                                        echo Waiting for 1 seconds...
+                                        sleep 1
+                                        JOBS_TO_GO=$(php w/maintenance/showJobs.php | tr -d '[:space:]')
+                                        echo $JOBS_TO_GO jobs to go
+                                    done
+                                    CMD
+                                ],
+                            ]
+                        ],
+                        'restartPolicy' => 'Never'
+                    ]
+                ]
+            ]
+        ]);
+        $jobName = $kubernetesClient->jobs()->create($jobSpec)['metadata']['name'];
+
+        // TODO: figure out how to use `wait` method instead
+        while (true) {
```

Comment:
This is kind of ugly :tm: but it works. There is a `watch` method provided by the Kubernetes client, but it has zero documentation and my PHP knowledge is not good enough to use it in a blocking manner. If someone knows how to use it here, any help is appreciated. This issue https://github.com/maclof/kubernetes-client/issues/58 is the only hint on how it could be used that I found.

Reply:
What would you think about not running this in a blocking way but simply having the API submit the job and then move on?

We could consider throttling the number of concurrent jobs being flushed through with a solution inside kubernetes like this: https://stackoverflow.com/a/71069007

I think this may be better since a badly behaving job on one wiki is less likely to block up everything and the stability of the api queue instance is less essential to keeping the concurrency controlled (e.g. imagine due to some bug the api queue is regularly restarting we could end up with many jobs being submitted at the same time since really the lock is only held per running process)

Reply:
I would guess (crudely) that by including the wiki in the name of the job this would deduplicate (e.g. stop using generate name and use a defined name)

Reply:
Yeah, I think I like that.